### PR TITLE
fix(fxFlexOrder): remove order after responsive deactivation

### LINKED ISSUE #296: feat(breakpoints): only apply flex properties if a breakpoint is activated
(opening issue per @ThomasBurleson suggestion in #201)

My comment:

> I'd like to be able disable flex layout completely for some elements when the width is extra small. But right now, even if I use gt-xs breakpoints, flex CSS properties are being added to the elements.
>
> 
> ```html
> <div fxLayoutAlign.gt-xs="center" fxLayout.gt-xs="row" fxLayoutGap.gt-xs="25px">
>   <div fxFlex.gt-xs="500px" class="victor"></div>
> </div>
> ```
>
> The layout element has `display: flex; flex-direction: row;` and the "victor" element has `flex: 1 1 0 0.00000001px;` when the width is xs. I expected that using only `gt-xs`, and no defaults, would mean that the flex properties would only be applied when the screen was greater than xs.

For further context, the main motivation is for a responsive layout where I want rows to be columns (stacked) on mobile. Safari has requirement of parent column containers having a set height (not auto), or else elements start clobbering each other. I want the child elements to be `display: block; height: auto;`.

Thanks :)

### LINKED ISSUE #697: fxLayoutGap not working with fxLayout fxFlexOrder
#### Bug:
fxLayoutGap does not work when fxFlexOrder is given with responsive breakpoints.Initially it is rendered properly. But when the screen is resized,fxLayoutGap does not work.

#### What is the expected behavior?
fxLayoutGap should work when screen is resized.

#### What are the steps to reproduce?
[stackblitz](https://stackblitz.com/edit/angular-flex-layout-seed-xseaux?file=app/test.component.ts)

1. Initial Page rendered.

image

2. After page resize
image

3. normal mode after resizing.

image

### PR DESCRIPTION
**Pull request by CaerusKaru**
Fixes #697 
Fixes #296

### CODE REVIEW

File: src/lib/core/base/base2.ts [hunk 1]
```diff
@@ -18,6 +18,9 @@ export abstract class BaseDirective2 implements OnChanges, OnDestroy {
   protected DIRECTIVE_KEY = '';
   protected inputs: string[] = [];
   protected destroySubject: Subject<void> = new Subject();
+  protected observables: Observable<any>[] = [];
```
Comment (ThomasBurleson) [inline src/lib/core/base/base2.ts]: Can we find a better name for this?

File: src/lib/core/base/base2.ts [hunk 2]
```diff
@@ -78,10 +86,20 @@ export abstract class BaseDirective2 implements OnChanges, OnDestroy {
       }
     }
 
+    this.mru = {...genStyles};
     this.applyStyleToElement(genStyles);
     builder.sideEffect(input, genStyles, parent);
   }
 
+  /** Remove generated styles from an element using predefined style builder */
+  protected clearStyles() {
```
Comment (ThomasBurleson) [inline src/lib/core/base/base2.ts]: Under what conditions should `clearStyles()` be called?

Comment (CaerusKaru) [reply src/lib/core/base/base2.ts]: When there isn't fallback value for a given element and key, it indicates that there is no actual value for the activated breakpoints, meaning no styles should be applied. To accomplish this, we clear out the most recently used styles by calling this method.

File: src/lib/core/base/base2.ts (hunk 1 above)
Comment (CaerusKaru) [reply src/lib/core/base/base2.ts]: How about `triggerObservables`?

**Comment by ThomasBurleson** [comment]
closed for https://github.com/angular/flex-layout/pull/926

**Pull request closed without merging**